src/qibolab/instruments/rfsoc/driver.py

```diff
@@ -368,20 +357,20 @@ def get_if_python_sweep(self, sequence: PulseSequence, *sweepers: rfsoc.Sweeper)
             loop, false otherwise
         """
         for sweeper in sweepers:
-            for sweep_idx, parameter in enumerate(sweeper.parameters):
-                if parameter is rfsoc.Parameter.BIAS:
-                    continue
-                if parameter is rfsoc.Parameter.DURATION:
-                    return True
+            if all(parameter is rfsoc.Parameter.BIAS for parameter in sweeper.parameters):
+                continue
+            if all(parameter is rfsoc.Parameter.DELAY for parameter in sweeper.parameters):
+                continue
+            if any(parameter is rfsoc.Parameter.DURATION for parameter in sweeper.parameters):
+                return True
```

Comment:
To avoid repeated iterations, maybe collect unique instances first:
```suggestion
            params = set(sweeper.parameters)
            if rfsoc.Parameter.DURATION in params:
                return True
            if len(params) == 1:
                if rfsoc.Parameter.DELAY in params or rfsoc.Parameter.DURATION in params:
                    continue
```

True that in practice this list won't be long. If you prefer to keep as is for readability, it is perfectly fine.

Reply:
I think as it is now it's a bit better for readability since it "sounds more like the rule itself".
But I have a question on your proposal, why did you do `if len(params) == 1`? I am not doing that right?

Reply:
> But I have a question on your proposal, why did you do `if len(params) == 1`? I am not doing that right?

They have to be `all()` the same. In order to check that, I was looking for a single unique value (and then I'm looking if the value is of the specified kind).

Reply:
Oh, I see... I forgot that `set` is of uniques... I can implement it

Reply:
> Oh, I see... I forgot that `set` is of uniques... I can implement it

No need: as you said above, it's more readable the original one. Performances here are not critical, so the readable version is fine :)